cobc/parser.y

```diff
@@ -7305,43 +7305,88 @@ volatile_clause:
 
 picture_clause:
   PICTURE	/* token from scanner, includes full picture definition */
-  _pic_locale_format
   {
 	check_repeated ("PICTURE", SYN_CLAUSE_4, &check_pic_duplicate);
 	current_field->pic = CB_PICTURE ($1);	/* always returned, invalid picture will have size == 0 */
-
-	if (CB_VALID_TREE ($2)) {
-		if (  (current_field->pic->category != CB_CATEGORY_NUMERIC
-		    && current_field->pic->category != CB_CATEGORY_NUMERIC_EDITED)
-		 || strpbrk (current_field->pic->orig, " CRDB-*") /* the standard seems to forbid also ',' */) {
-			cb_error_x ($1, _("a locale-format PICTURE string must only consist of '9', '.', '+', 'Z' and the currency-sign"));
-		} else {
-			/* TODO: check that not we're not within a CONSTANT RECORD */
-			CB_PENDING_X ($1, "locale-format PICTURE");
-		}
-	}
+  }
+  _pic_locale_format_or_depending_on
+  {
+	  if ((!current_field->pic || current_field->pic->variable_length) &&
+	      !current_field->flag_picture_l) {
+		  /* Current field with PIC L was not translated */
+		  cb_error_x (CB_TREE (current_field->pic),
+			      _("missing DEPENDING clause for PICTURE string with "
+				"'L' character"));
+	  }
   }
 ;
 
-_pic_locale_format:
+_pic_locale_format_or_depending_on:
   /* empty */
-  { $$ = NULL; }
-| LOCALE _is_locale_name SIZE _is integer  
+| LOCALE _is_locale_name SIZE _is integer
+  {
+	  /* $2 -> optional locale-name to be used */
+	  if (CB_VALID_TREE ($5)) {
```

Comment:
The valid_tree check for 5 (the integer) can be dropped (that complete if here); using $1 (the LOCALE token) below for PENDING seems to be troublesome, maybe drop the `_X ($1, `?